src/pages/judge/[secret].tsx

```diff
@@ -1,236 +1,239 @@
+        setPreviousTeam(pTeam);
+      } else {
+        setPreviousTeam(null);
+      }
+    } else {
+      setCurrentTeam(null);
+      setPreviousTeam(null);
+    }
+    setLoading(false);
+  };
+
+  if (!startJudging) {
+    return (
+      <div className="text-center mt-5">
+        <img className="mb-4" src="/logo.png" />
+        <h4 className="text-center">Make sure to open this in a browser (not through an app)</h4>
+        <button className="btn btn-primary" onClick={(): void => setStartJudging(true)}>
+          Start Judging
+        </button>
+      </div>
+    );
+  }
+
+  if (loading) {
+    return <h1>Loading</h1>;
+  }
+
+  if (doneJudging) {
+    return (
+      <>
+        <h1 style={{ marginTop: 50 }} className="text-center">
+          Congratulations!
+        </h1>
+        <h4 className="text-center">You are a supreme judger</h4>
+        <h5 className="text-center">You&apos;re done with judging</h5>
+        <h6 className="text-center">Thanks</h6>
+      </>
+    );
+  }
+
+  return (
+    <div className="container-fluid text-center">
+      <h1>Welcome Judge!</h1>
+      {error && <p style={{ color: 'red' }}>{error}, refresh and try again</p>}
+      {currentTeam && (
+        <div className="alert alert-info text-left" role="alert">
+          <h3>Current Team</h3>
+          <h4>{currentTeam.name}</h4>
+          <p>
+            <strong>Location: {currentTeam.tableNumber}</strong>
+          </p>
+          <p>{currentTeam.projectDescription}</p>
+        </div>
+      )}
+      {previousTeam && (
+        <div className="alert alert-info text-left" role="alert">
+          <h3>Previous Team</h3>
+          <h4>{previousTeam.name}</h4>
+          <p>
+            <strong>Location: {previousTeam.tableNumber}</strong>
+          </p>
+          <p>{previousTeam.projectDescription}</p>
+        </div>
+      )}
+      {firstTeam && <h3>Once you&apos;ve judged the current team, press continue</h3>}
+      {firstTeam && (
+        <button type="button" className="btn btn-primary btn-lg" onClick={voteHandler()}>
+          Next Team
+        </button>
+      )}
+      {!firstTeam && (
+        <>
+          <h3>Which team was better?</h3>
+          <div className="btn-group" role="group">
+            <button type="button" className="btn btn-primary" onClick={voteHandler(false)}>
+              Previous Team
+            </button>
+            <button type="button" className="btn btn-default" onClick={skipHandler()}>
+              Skip Team
+            </button>
+            <button type="button" className="btn btn-primary" onClick={voteHandler(true)}>
+              Current Team
+            </button>
+          </div>
+        </>
+      )}
+    </div>
+  );
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+judge.getInitialProps = async (ctx: any): Promise<any> => {
+  const judgeSecret = await Config.findOneOrFail('judgeSecret');
+  if (ctx.res && ctx.query.secret !== judgeSecret.value) {

```

Comment:
`ctx.res` will only be there on the server, so we're basically saying if we're on the server and the secret does not match, throw a 404. Because of that we only want to reach out to the database if we're on the server.

```suggestion
  if (ctx.res && ctx.query.secret !== judgeSecret.value) {
  const judgeSecret = await Config.findOneOrFail('judgeSecret');
```

Reply:
We're confused on this suggestion because if we switch the two lines then we get an error since judgeSecret.value is not yet defined. Also, if judgeSecret.value is defined only within the if statement then it messes up the return statement as well since it is not defined outside of the if statement.